test/state/test_state.hpp

```diff
@@ -0,0 +1,48 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include <evmc/evmc.hpp>
+#include <intx/intx.hpp>
+#include <unordered_map>
+
+namespace evmone
+{
+namespace state
+{
+class State;
+}
+
+namespace test
+{
+using evmc::address;
+using evmc::bytes;
+using evmc::bytes32;
+using intx::uint256;
+
+// TestAccount struct based on state::Account
+struct TestAccount
+{
+    uint64_t nonce = 0;
+    uint256 balance;
+    std::unordered_map<bytes32, bytes32> storage;
+    bytes code;
+
+    bool operator==(const TestAccount&) const noexcept = default;
+};
+
+class TestState : public std::unordered_map<address, TestAccount>
```

Comment:
maybe we could summarize in a single sentence, how is it different from `State`? Just the gist of it. Since both are kept in the `test/` subdir, they are both, in a way, test-states, so it might be worth explaining when to expect which

Reply:
Added comments.